src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_QueryLambdaRewriter.vb

```diff
@@ -0,0 +1,93 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
+
+Namespace Microsoft.CodeAnalysis.Semantics
+    Partial Friend NotInheritable Class VisualBasicOperationFactory
+        Private Shared Function RewriteQueryLambda(node As BoundQueryLambda) As BoundNode
+            ' We rewrite query lambda into regular lambda with 2 passes.
+            ' Pass 1 uses helper methods from LocalRewriter to do the lowering. This introduces large number of DAGs.
+            ' Pass 2 walks over the lowered tree and replaces duplicate bound nodes in the tree with their clones - this is a requirement for the Operation tree.
+            Dim pass1Rewriter As New QueryLambdaRewriterPass1
+            Dim rewrittenLambda As BoundLambda = DirectCast(pass1Rewriter.VisitQueryLambda(node), BoundLambda)
```

Comment:
>pass1Rewriter.VisitQueryLambda(node) [](start = 60, length = 36)

It looks like this is also going to rewrite all query lambdas nested into the body of this lambda. This is probably fine. Consider adding a comment to that effect.